ReactiveObjC/RACTuple.h

```diff
@@ -111,15 +115,84 @@ NS_ASSUME_NONNULL_BEGIN
 - (nullable id)objectAtIndexedSubscript:(NSUInteger)idx;
 @end
 
+/// A tuple with exactly two generic values.
+@interface RACTwoTuple<__covariant First, __covariant Second> : RACTuple
```

Comment:
How about this as a possible use case? Say we have a protocol which defines a method returning a `RACCommand` of a variable-size tuple, like so:

```objc
@protocol MyProtocol
- (RACCommand<RACTuple *, RACUnit *> *)myCommand;
@end
```

Currently we might implement it like so:

```objc
// One implementation.
- (RACCommand<RACTuple *, RACUnit *> *)myCommand {
    return [[RACCommand alloc] initWithSignalBlock:^RACSignal * _Nonnull(RACTuple * _Nullable oneValueTuple) {
        // Do something
        return [RACSignal return:[RACUnit defaultUnit]];
    }];
}

// Another implementation.
- (RACCommand<RACTuple *, RACUnit *> *)myCommand {
    return [[RACCommand alloc] initWithSignalBlock:^RACSignal * _Nonnull(RACTuple * _Nullable twoValueTuple) {
        // Do something
        return [RACSignal return:[RACUnit defaultUnit]];
    }];
}
```

But unless I'm mistaken, it should now be possible to implement it in the following way:



```objc
// One implementation -- .h
- (RACCommand<RACOneTuple<NSString *> *, RACUnit *> *)myCommand;

// .m
- (RACCommand<RACOneTuple<NSString *> *, RACUnit *> *)myCommand {
    return [[RACCommand alloc] initWithSignalBlock:^RACSignal * _Nonnull(RACOneTuple<NSString *> * _Nullable oneValueTuple) {
        // Do something
        return [RACSignal return:[RACUnit defaultUnit]];
    }];
}

// Another implementation -- .h
- (RACCommand<RACTwoTuple<NSString *, NSNumber *> *, RACUnit *> *)myCommand;

// .m
- (RACCommand<RACTwoTuple<NSString *, NSNumber *> *, RACUnit *> *)myCommand {
    return [[RACCommand alloc] initWithSignalBlock:^RACSignal * _Nonnull(RACTwoTuple<NSString *, NSNumber *> * _Nullable twoValueTuple) {
        // Do something
        return [RACSignal return:[RACUnit defaultUnit]];
    }];
}
```

So in that case it would be useful to have `RACOneTuple` because then we can specify the type of its value. It would also be useful, although not as useful, to have `RACZeroTuple` in case we want to implement `-myCommand` with no arguments (where I suppose we'd currently use `[RACTuple new]` and then just not use the argument).